packages/console/src/components/OrganizationRolePermissionsAssignmentModal/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { type AdminConsoleKey } from '@logto/phrases';
+import { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import ConfirmModal from '@/ds-components/ConfirmModal';
+import DataTransferBox from '@/ds-components/DataTransferBox';
+import TabNav, { TabNavItem } from '@/ds-components/TabNav';
+import TabWrapper from '@/ds-components/TabWrapper';
+
+import { PermissionType } from './types';
+import useOrganizationRolePermissionsAssignment from './use-organization-role-permissions-assignment';
+
+const permissionTabs = {
+  [PermissionType.Organization]: {
+    title: 'organization_role_details.permissions.organization_permissions',
+    key: PermissionType.Organization,
+  },
+  [PermissionType.Api]: {
+    title: 'organization_role_details.permissions.api_permissions',
+    key: PermissionType.Api,
+  },
+} satisfies {
+  [key in PermissionType]: {
+    title: AdminConsoleKey;
+    key: key;
+  };
+};
+
+type Props = {
+  organizationRoleId: string;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+function OrganizationRolePermissionsAssignmentModal({
+  organizationRoleId,
+  isOpen,
+  onClose,
+}: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const {
+    activeTab,
+    setActiveTab,
+    onSubmit,
+    organizationScopesAssignment,
+    resourceScopesAssignment,
+    clearSelectedData,
+    isLoading,
+  } = useOrganizationRolePermissionsAssignment(organizationRoleId);
+
+  const onCloseHandler = useCallback(() => {
+    onClose();
+    clearSelectedData();
+    setActiveTab(PermissionType.Organization);
+  }, [clearSelectedData, onClose, setActiveTab]);
+
+  const onSubmitHandler = useCallback(async () => {
+    await onSubmit();
```

Comment:
Do we have loading on the button when the submit post request is in progress?

Reply:
Yes, we do.
<img width="707" alt="image" src="https://github.com/logto-io/logto/assets/10806653/37e5c0c0-c7f0-4ab4-b41d-2a9dbaf67e35">